runtime/Makefile

```diff
@@ -79,6 +80,7 @@ configure:
 		-DENABLE_LIGHTNING_KOKKOS=$(ENABLE_LIGHTNING_KOKKOS) \
 		-DENABLE_OPENQASM=$(ENABLE_OPENQASM) \
 		-DENABLE_OPENMP=$(LIGHTNING_ENABLE_OPENMP) \
+		-DKokkos_ENABLE_CUDA=$(KOKKOS_ENABLE_CUDA) \
```

Comment:
I thought it would be nicer in the long run to just pass all Kokkos options through `CMAKE_ARGS`, otherwise we need to introduce several options that effectively just rename these options, and I would prefer not introducing an option for each parallel model supported by Kokkos. As for `Kokkos_ARCH_*`, this is usually detected correctly by CMake if the node used to build has the same GPUs as the nodes used to run. Otherwise, this is required (for SYCL, HIP as well) and we do not want to duplicate all these options for sure. 

I just thought this would have the lowest impact for the time being, but I'm happy to change it and adapt the CI workflows as needed here.